### PR TITLE
Add batch price update instruction to Oracle

### PR DESCRIPTION
**Pull request by tompntn**
### Overview
This PR adds a new instruction to the Oracle: `cmd_upd_price_no_fail_on_error`. This is exactly the same as `cmd_upd_price`, but never fails the instruction, and therefore the transaction, even if the update is unsuccessful. This is because we will soon send update instructions in batches, and want to prevent a failure in one price feed blocking updates of other prices.

### Approach
- Add the `cmd_upd_price_no_fail_on_error` instruction.
- Change `pythd` to send `cmd_upd_price_no_fail_on_error` JPRC requests instead of `cmd_upd_price`.

### Testing
- `pythd` changes: the existing `pythd` integration tests for the JRPC `update_price` request will now cover the new `e_cmd_upd_price_no_fail_on_error` code paths.
- Oracle changes: new Oracle tests for the `cmd_upd_price_no_fail_on_error` instructions have been added, demonstrating the optimistic failure semantics.

### Caveats
All price update transactions will appear on-chain, even if they are unsuccessful. This means publishers will have to rely on [inspecting publishing slots](https://github.com/pyth-network/pyth-client/blob/main/doc/publish_slot.md) to be sure if their updates were successful.

### CODE REVIEW

File: program/src/oracle/oracle.c [hunk 1]
```diff
@@ -475,68 +475,156 @@ static uint64_t upd_test( SolParameters *prm, SolAccountInfo *ka )
...
+static uint64_t upd_price( SolParameters *prm, SolAccountInfo *ka )
+{
+  // Validate command parameters
+  cmd_upd_price_t *cptr = (cmd_upd_price_t*)prm->data;
+  if ( prm->data_len != sizeof( cmd_upd_price_t ) ) {
+    return ERROR_INVALID_ARGUMENT;
+  }
+
+  // Verify that the correct number of accounts have been passed
+  if ( (prm->ka_num != 3 && prm->ka_num != 4) ) {
```
[the remark this answers is not in the document]
Comment (tompntn) [reply program/src/oracle/oracle.c:605]: Good question - I'm not sure. AFAICT `pythd` only ever sends 3 accounts, and the Oracle program doesn't ever use the 4th one. I can't see anywhere this is used, but maybe I'm missing something.

**Review by jayantk — APPROVED** [review]
Nice this should definitely work. Again would still like @Reisen to review before merging.

**Review by jayantk — APPROVED** [review]
well, at this point, it's a tiny change, so feel free to merge even without @Reisen

File: program/src/oracle/oracle.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (dblackman0) [reply program/src/oracle/oracle.c]: I think an old version of the onchain program took a 'parameter' account which was removed in https://github.com/pyth-network/pyth-client/commit/cd9dc8c60ea30851b5803d192c27a13f170a959e

**Pull request merged**